firefox/background.entry.js

```diff
@@ -138,7 +138,19 @@ let db;
 	}
 }
 
-addListener('storage', ([operation, key, value]) => {
+function storageFailureAlert(worker) {
+	sendMessage('alert', `
+		<p><b>An error occurred while creating the IndexedDB database.</b></p>
+		<p>Reddit Enhancement Suite will not function.</p>
+		<p>Your Firefox profile may be corrupted (<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=944918">Bug 944918</a>).</p>
+		<br>
+		<p>Please report this to the relevant beta thread or /r/RESissues.</p>
+	`, worker);
+}
+
+addListener('storage', ([operation, key, value], worker) => {
+	if (!db) storageFailureAlert(worker);
```

Comment:
Three or four storage requests are made before init crashes, so this will open multiple alerts in series.

But I don't think that's a problem because you're going to be immediately rolling back/disabling the extension if this happens (memoizing by worker and current URL would be mildly complex for little gain).